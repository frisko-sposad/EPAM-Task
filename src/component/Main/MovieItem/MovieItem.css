.movie-item {  
  color: gray;
  margin: 1rem;  
}

.movie-item__img {
  width: 300px;
}

.movie-item__info {
  display: flex;  
  justify-content: space-between;
  padding: 0.3rem 0.5rem;
}

.movie-item__year {
  font-size: 0.8rem;
  padding: 0 0.4rem;
  border: 1px solid gray;
}
