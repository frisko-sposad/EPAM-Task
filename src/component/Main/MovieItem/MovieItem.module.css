.MovieItem {  
  color: gray;
  margin: 1rem;  
}

.MovieItem__img {
  width: 300px;
}

.MovieItem__info {
  display: flex;  
  justify-content: space-between;
  padding: 0.3rem 0.5rem;
}

.MovieItem__year {
  font-size: 0.8rem;
  padding: 0 0.4rem;
  border: 1px solid gray;
}
