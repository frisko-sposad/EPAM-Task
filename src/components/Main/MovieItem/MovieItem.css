.movie-item {
  color: gray;
  margin: 1rem;
  width: 300px;
  height: 505px;
}

.movie-item__img {
  width: 300px;
  cursor: pointer;
}

.movie-item__info {
  display: flex;
  font-size: 0.8rem;
  justify-content: space-between;
  padding: 0.3rem 0.5rem;
}

.movie-item__title {
  font-size: 0.9rem;
  color: black;
  text-transform: uppercase;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.movie-item__year {
  font-size: 0.6rem;  
  padding: 0 0.4rem;
  white-space: nowrap;
  border: 1px solid gray;
}
