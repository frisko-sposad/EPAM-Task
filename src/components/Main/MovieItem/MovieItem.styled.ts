import styled from 'styled-components';

export const MovieItemContainer = styled.div`
  color: gray;
  margin: 1rem;
  width: 300px;
  height: 532px;
  cursor: pointer;
`;

export const Image = styled.img`
  width: 300px;
  height: 475px;
`;

export const Info = styled.div`
  display: flex;
  font-size: 0.8rem;
  justify-content: space-between;
  align-items: center;
  padding: 0.3rem 0.5rem;
`;

export const Title = styled.div`
  display: flex;
  font-size: 0.8rem;
  justify-content: space-between;
  align-items: center;
  padding: 0.3rem 0.5rem;
`;

export const Year = styled.div`
  font-size: 0.6rem;
  padding: 0.2rem 0.4rem;
  white-space: nowrap;
  border: 1px solid gray;
`;
