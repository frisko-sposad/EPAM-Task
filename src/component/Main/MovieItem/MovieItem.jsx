import React from 'react';
import './MovieItem.css';

const MovieItem = ({ image, name, year, genre }) => (
  <div className="movie-item">
    <img className="movie-item__img" src={image} alt="movieImage" />
    <div>
      <div className="movie-item__info">
        <div>{name}</div>
        <div className="movie-item__year">{year}</div>
      </div>
      <div className="movie-item__info">
        <div>{genre}</div>
      </div>
    </div>
  </div>
);

export default MovieItem;
