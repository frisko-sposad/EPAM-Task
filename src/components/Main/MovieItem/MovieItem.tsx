import React, { FC } from 'react';
import './MovieItem.css';

export interface MovieItemProps {
  image: string;
  name: string;
  year: string;
  genre: string;
}

const MovieItem: FC<MovieItemProps> = ({ image, name, year, genre }) => (
  <div className="movie-item">
    <img className="movie-item__img" src={image} alt="movieImage" />
    <div>
      <div className="movie-item__info">
        <div>{name}</div>
        <div className="movie-item__year">{year}</div>
      </div>
      <div className="movie-item__info">{genre}</div>
    </div>
  </div>
);

export default MovieItem;
