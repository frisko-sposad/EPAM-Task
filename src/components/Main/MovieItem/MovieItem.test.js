import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import MovieItem from './MovieItem';
import movieBase from '../../MovieBase';

describe('Render MovieItem.', () => {
  it('MovieItem', () => {
    const component = mount(
      <BrowserRouter>
        <MovieItem {...movieBase[0]} />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
