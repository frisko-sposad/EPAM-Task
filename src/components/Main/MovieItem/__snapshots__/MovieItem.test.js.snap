// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieItem. MovieItem 1`] = `
<MovieItem
  genres={
    Array [
      "fantasy",
      "Drama",
    ]
  }
  id={1}
  posterPath="https://avatars.mds.yandex.net/get-kinopoisk-image/1704946/57375a05-4883-4994-b33b-8d60741a0706/300x450"
  releaseDate="2019"
  title="The Witcher"
>
  <Link
    href="/film/1"
    passHref={true}
  >
    <a
      href="/film/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <MovieItemstyled__MovieItemContainer>
        <div
          className="MovieItemstyled__MovieItemContainer-sc-1igl0l9-0 cCNLhl"
        >
          <MovieItemstyled__Image
            alt="poster from The Witcher"
            src="https://avatars.mds.yandex.net/get-kinopoisk-image/1704946/57375a05-4883-4994-b33b-8d60741a0706/300x450"
          >
            <img
              alt="poster from The Witcher"
              className="MovieItemstyled__Image-sc-1igl0l9-1 HEkJa"
              src="https://avatars.mds.yandex.net/get-kinopoisk-image/1704946/57375a05-4883-4994-b33b-8d60741a0706/300x450"
            />
          </MovieItemstyled__Image>
          <div>
            <MovieItemstyled__Info>
              <div
                className="MovieItemstyled__Info-sc-1igl0l9-2 FjQtw"
              >
                <MovieItemstyled__Title>
                  <div
                    className="MovieItemstyled__Title-sc-1igl0l9-3 gGwrCN"
                  >
                    The Witcher
                  </div>
                </MovieItemstyled__Title>
                <MovieItemstyled__Year>
                  <div
                    className="MovieItemstyled__Year-sc-1igl0l9-4 hksARX"
                  >
                    2019
                  </div>
                </MovieItemstyled__Year>
              </div>
            </MovieItemstyled__Info>
            <MovieItemstyled__Info>
              <div
                className="MovieItemstyled__Info-sc-1igl0l9-2 FjQtw"
              >
                fantasy, Drama
              </div>
            </MovieItemstyled__Info>
          </div>
        </div>
      </MovieItemstyled__MovieItemContainer>
    </a>
  </Link>
</MovieItem>
`;
